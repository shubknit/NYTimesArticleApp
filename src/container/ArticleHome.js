import React, { Component } from 'react';
import { View, ActivityIndicator } from 'react-native';
import { getArticlesData } from '../services/NYTimesAPI';
import  { ArticleList }  from '../components/ArticleList';


/* Component rendering Article list fetching from NyTimesAPI */

export class ArticleHome extends Component {
  constructor(){
    super();
    this.state = {
      articlesData: {},
      isLoading: true 
    }
  }
  
  componentDidMount(){
    this.getArticlesData();
  }

  /* Handling response from Nytimes API */

  handleResponse = (response) => {
    this.setState({
      articlesData: response,
      isLoading: false
    })
  }

  /* Handling response from Nytimes API */

  handleError = (e) => {
    console.log('error in fetching',e)
  }

  /* Fetching data from NyTimess service */

  getArticlesData = async () => {
    const response = await getArticlesData().catch(e => {
      this.handleError(e);
    })
    this.handleResponse(response);
  }

  render(){
    const { results } = this.state.articlesData;
    const { navigate } = this.props.navigation;
    const loaderView = <ActivityIndicator size="large" color="#00ff00" />;
    const articleListView =  <ArticleList articlesData = {results} navigate = {navigate}/>;
    return (
      <View>
        { this.state.isLoading ? loaderView : articleListView } 
      </View>
     
    )
  }
}