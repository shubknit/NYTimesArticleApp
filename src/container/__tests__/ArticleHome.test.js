import React from 'react';
import { ArticleHome } from '../ArticleHome';
import renderer from 'react-test-renderer';

/* Test for Article Home component */

const navigation = {
    navigate: {}
}

/* Test for Article Home component Rendering */

it('should render the Article Home Page correctly', () => {
    const wrapper = renderer.create(<ArticleHome navigation = {navigation}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})