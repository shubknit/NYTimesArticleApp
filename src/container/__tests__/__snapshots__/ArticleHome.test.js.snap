// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Article Home Page correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#00ff00"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
